bundle/config/validate/job_cluster_key_defined.go

```diff
@@ -0,0 +1,66 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+)
+
+func JobClusterKeyDefined() bundle.Mutator {
+	return &jobClusterKeyDefined{}
+}
+
+type jobClusterKeyDefined struct {
+}
+
+func (v *jobClusterKeyDefined) Name() string {
+	return "validate:job_cluster_key_defined"
+}
+
+func (v *jobClusterKeyDefined) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	// Collect all job_cluster_key references from defined tasks
+	jobClusterKeys := make(map[string]location)
+	for i, job := range b.Config.Resources.Jobs {
+		for j, task := range job.Tasks {
+			if task.JobClusterKey != "" {
```

Comment:
You could use `dyn.MapByPattern` on `resources.jobs.*.tasks[].job_cluster_key` (and the other one for for each tasks) and have direct access to the location.

Reply:
To synthesize the pattern you would need to use `dyn.AnyKey()` and `dyn.AnyIndex()`. There is no parsing of `*` or `[]` from a string (yet).